### ISSUE TITLE
add fetch-node: fetch is not defined

### ISSUE THREAD

**Issue by mohammed-gehad** [nextauthjs/next-auth#4624]
### Description 📓

i suggest adding fetch-node to the project to avoid this error 
[next-auth][error][CLIENT_FETCH_ERROR] 
ReferenceError: fetch is not defined

### How to reproduce ☕️

using in custom server

### Contributing 🙌🏽

Yes, I am willing to help implement this feature in a PR

**Comment by balazsorban44** [comment]
This is likely a project configuration issue. Next.js comes with a `fetch` polyfill. Your error is kinda hard to understand, as `CLIENT_FETCH_ERROR` comes from the browser, and `fetch` is a Web Standard API.

Hard to say more without a reproduction.

**Comment by mohammed-gehad** [comment]
yeah i understand that  Next.js comes with a fetch polyfill. 
but the problem occur when using with a custom server (eg express server). that is not a part of next.js where the environment doesn't have a fetch polyfill.
or next-auth should only be used with Next.js api routes?

**Comment by surajsharma** [comment]
getting the same error on this line
`const session = await getSession({ req });`

any updates?

**Comment by mohammed-bahumaish** [comment]
> 
i have solved this issue by polyfilling fetch using node-fetch
 
```
import fetch, { Headers, Request, Response } from 'node-fetch'

if (!globalThis.fetch) {
  globalThis.fetch = fetch
  globalThis.Headers = Headers
  globalThis.Request = Request
  globalThis.Response = Response
}
```

then in the context

```
  const token = await getToken({
    req,
    secret: process.env.NEXTAUTH_SECRET,
    cookieName: 'next-auth.session-token',
  })
```

**Comment by adityasajoo** [comment]
Did. you fix this issue ?

**Comment by surajsharma** [comment]
> Did you find a solution?

yup, turns out i was on an older node version, switched to 18.10.0 and it was gone

**Comment by mohammed-bahumaish** [comment]
> Did you find a solution?

this error occurs when global fetch is undefined.
solution: polyfill fetch with 'node-fetch' or any similar package.

**Comment by prabhjotsinghchadha** [comment]
I am also getting same error

**Comment by MehtaManan07** [comment]
> This is likely a project configuration issue. Next.js comes with a `fetch` polyfill. Your error is kinda hard to understand, as `CLIENT_FETCH_ERROR` comes from the browser, and `fetch` is a Web Standard API.

Were you able to solve it??

**Comment by MehtaManan07** [comment]
1. Navigate to the node_modules/next-auth/client directory
2. Open the _utils.js file
3. At the top of the file, import the node-fetch package by adding the following line:
`var nodeFetch = require('node-fetch');`
4. Replace all instances of fetch with nodeFetch. You will find the following line:
`return fetch(url, options);`
change it to
`return nodeFetch(url, options);`
5. Save the changes and you are good to go

**Comment by SashenJayathilaka** [comment]
Update your **Node js Latest Current Version**. After that I think this issue fix.

**Comment by Mayvis** [comment]
If you're using a version of Node prior to 18, the [fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) is not implemented out-of-the-box and you'll need to use an external module for that. like [node-fetch](https://github.com/bitinn/node-fetch), [isomorphic-fetch](https://www.npmjs.com/package/isomorphic-fetch)...etc.

**Comment by angellutz** [comment]
Hey all. Had this issue myself (fetch is not defined). Upgraded Node.js to version 18 as suggested here and it solved the problem. Thanks for the help!
